### PR TITLE
Warn when calling synchronized on AnyVal

### LINKED ISSUE #17284: Should Generate warnings when synchronized on `AnyVal`
## Compiler version
3.2.2

## Minimized code

```Scala
1.synchronized{}
```

## Output
image

## Expectation

generate some warning as https://openjdk.org/jeps/390

### PR DESCRIPTION
**Pull request by mbovel**
Fixes #17284. Done during the Dotty Issue Spree of June 20, 2023.

### CODE REVIEW

File: tests/neg/17284.scala [hunk 1]
```diff
@@ -0,0 +1 @@
+def test = 451.synchronized {}
```
Comment (nicolasstucki) [inline tests/neg/17284.scala:1]: This test should be testes with fatal warnings

Comment (nicolasstucki) [inline tests/neg/17284.scala:1]:
```suggestion
def test = 451.synchronized {} // error
```

Comment (mbovel) [reply tests/neg/17284.scala]: Yes, I committed an old version of the test file by mistake. Fixed.

Comment (mbovel) [reply tests/neg/17284.scala]: Same here

File: compiler/src/dotty/tools/dotc/reporting/messages.scala [hunk 2]
```diff
@@ -2345,6 +2345,15 @@ class UnqualifiedCallToAnyRefMethod(stat: untpd.Tree, method: Symbol)(using Cont
        |you intended."""
 }
 
+class SynchronizedCallOnBoxedClass(stat: untpd.Tree)(using Context)
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/reporting/messages.scala:2348]:
```suggestion
class SynchronizedCallOnBoxedClass(stat: tpd.Tree)(using Context)
```

**Review by nicolasstucki — COMMENTED** [review]
Otherwise LGTM. Should squash commits.

File: compiler/src/dotty/tools/dotc/reporting/messages.scala (hunk 2 above)
Comment (mbovel) [reply compiler/src/dotty/tools/dotc/reporting/messages.scala]: Committed ✅

**Comment by mbovel** [comment]
Rebased ✅

**Review by nicolasstucki — APPROVED** [review]

**Pull request merged**